arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/extensions/NumberInstancesTest.kt

```diff
@@ -21,18 +21,26 @@ import io.kotest.property.checkAll
 
 class NumberInstancesTest : UnitSpec() {
 
-  fun <F> testAllLaws(SG: Semiring<F>, M: Monoid<F>, GEN: Arb<F>) {
-    testLaws(SemiringLaws.laws(SG, GEN))
-    testLaws(MonoidLaws.laws(M, GEN))
+  fun <F> testAllLaws(
+    SG: Semiring<F>,
+    M: Monoid<F>,
+    GEN: Arb<F>,
+    eq: (F, F) -> Boolean = { a, b -> a == b }
+  ) {
+    testLaws(SemiringLaws.laws(SG, GEN, eq))
+    testLaws(MonoidLaws.laws(M, GEN, eq))
   }
 
   init {
     testAllLaws(Semiring.byte(), Monoid.byte(), Arb.byteSmall())
-    testAllLaws(Semiring.double(), Monoid.double(), Arb.doubleSmall())
-    testAllLaws(Semiring.int(), Monoid.int(), Arb.intSmall())
     testAllLaws(Semiring.short(), Monoid.short(), Arb.shortSmall())
-    testAllLaws(Semiring.float(), Monoid.float(), Arb.floatSmall())
+    testAllLaws(Semiring.int(), Monoid.int(), Arb.intSmall())
     testAllLaws(Semiring.long(), Monoid.long(), Arb.longSmall())
+    MonoidLaws.laws(Monoid.float(), Arb.floatSmall(), Float::eqv)
+    // TODO Semiring laws failing with == or Float::eqv
+    // testAllLaws(Semiring.float(), Monoid.float(), Arb.floatSmall(), Float::eqv)
+    // TODO Semiring/Monoid laws failing with == or Double::eqv
+    // testAllLaws(Semiring.double(), Monoid.double(), Arb.doubleSmall(), Double::eqv)
```

Comment:
Laws for `Semiring` and `Monoid` are failing due to the comparisons made.
Failures such as `0.0 neq -0.0`. Should we remove Semiring, Monoid for both `Float` and `Double` due to how they are compared by default?

The `Semigroup` is mostly also broken, but not caught by a law.

Reply:
I think in this case we should completely remove those instances they won't be lawful under kotlin's equality.

Reply:
Totally agree, I will deprecate the instances for `Float` & `Double` so we can remove these tests.